homeassistant/components/camera/__init__.py

```diff
@@ -197,6 +200,10 @@ def _get_camera_from_entity_id(hass, entity_id):
     component = hass.data[DOMAIN] = \
         EntityComponent(_LOGGER, DOMAIN, hass, SCAN_INTERVAL)
 
+    prefs = CameraPreferences(hass)
+    await prefs.async_initialize()
+    hass.data[DATA_CAMERA_PREFS] = prefs
```

Comment:
I would generally put the storage object on hass.data so that the websocket methods can use it, but that's not available if EntityComponent is put there.
A cursory glance shows that none of the platforms use the EntityComponent directly, so I'm not sure if it really needs to be on hass.data, but I don't have the historical context as to why the above pattern existed in the first place...

Reply:
We put entity component in hass.data so we can forward config entries. 

It's fine to take other keys than `DOMAIN`, as long as you make sure the key is prefixed with `camera_`